Add L2/L8 plots and extend summary info/plots with L5 offsets, L8 trims, and L9 MDP

bbeny123 opened the pull request:
This MR introduces two functional updates:

### 1. `--plot-type` option for `plot`

Adds support for plotting based on `L2` and `L8` metadata.

```
  -p, --plot-type <PLOT_TYPE>
          Sets the DV metadata level to plot

          [default: l1]

          Possible values:
          - l1:  L1 Dynamic Brightness
          - l2:  L2 Trims
          - l8t: L8 Trims (CM v4.0 RPU required)
          - l8s: L8 Saturation Vectors (CM v4.0 RPU required)
          - l8h: L8 Hue Vectors (CM v4.0 RPU required)
```

### 2. Extended metadata info in summary and plots

Additional metadata fields are now included in both the `info -s` and `plot` annotations:
- **L5 offsets**
- **L8 trims** (for `CM v4.0` **RPUs** only)
- **L9 MDP** (for `CM v4.0` **RPUs** only)

### Examples: 

#### 1. `info -s`

Current **CM v2.9**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 1 (CM v2.9)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 10000.00 nits, MaxFALL: 43.04 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 0 nits, MaxFALL: 0 nits
```

Current **CM v4.0**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 2 (CM v4.0)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 1023.27 nits, MaxFALL: 92.36 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 1023 nits, MaxFALL: 92 nits
  L2 trims: 100 nits
```

New **CM v2.9 + L5 0**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 1 (CM v2.9)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 10000.00 nits, MaxFALL: 43.04 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 0 nits, MaxFALL: 0 nits
  L5 offsets: TOP: 0, BOTTOM: 0, LEFT: 0, RIGHT: 0
```

New **CM v2.9 + L5 Standard**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 1 (CM v2.9)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 10000.00 nits, MaxFALL: 43.04 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 0 nits, MaxFALL: 0 nits
  L5 offsets: TOP: 42, BOTTOM: 42, LEFT: 0, RIGHT: 0
```

New **CM v2.9 + L5 Variable Top/Bottom**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 1 (CM v2.9)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 10000.00 nits, MaxFALL: 43.04 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 0 nits, MaxFALL: 0 nits
  L5 offsets: TOP: (0 - 280), BOTTOM: (0 - 280), LEFT: 0, RIGHT: 0
```

New **CM v4.0 + L5 Null**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 2 (CM v4.0)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 1023.27 nits, MaxFALL: 92.36 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 1023 nits, MaxFALL: 92 nits
  L5 offsets: TOP: N/A, BOTTOM: N/A, LEFT: N/A, RIGHT: N/A
  L2 trims: 100 nits
  L8 trims: 100 nits
  L9 MDP: DCI-P3 D65
```

New **CM v4.0 + L5 Variable**:
```
Summary:
  Frames: 147955
  Profile: 8
  DM version: 2 (CM v4.0)
  Scene/shot count: 1348
  RPU mastering display: 0.0001/1000 nits
  RPU content light level (L1): MaxCLL: 1023.27 nits, MaxFALL: 92.36 nits
  L6 metadata: Mastering display: 0.0001/1000 nits. MaxCLL: 1023 nits, MaxFALL: 92 nits
  L5 offsets: TOP: (0 - 280), BOTTOM: (0 - 280), LEFT: (0 - 60), RIGHT: (0 - 60)
  L2 trims: 100 nits
  L8 trims: 100 nits
  L9 MDP: DCI-P3 D65
```

#### 2. `plot`

Current **CM v2.9**:
![L1_CM29](https://github.com/user-attachments/assets/39460b82-63d6-4b4f-bc11-33fce157e56f)

Current **CM v4.0**:
![L1_CM40](https://github.com/user-attachments/assets/45e81c3f-3c44-4f60-a6e2-024ae6d51b40)

New **CM v2.9 + L5 0**:
![L1_CM29_L5-0](https://github.com/user-attachments/assets/5999cafa-c339-4091-b10b-839dc490aa58)

New **CM v2.9 + L5 Variable Top/Bottom**:
![L1_CM29_L5-VARIABLE-TB](https://github.com/user-attachments/assets/95d2378f-600e-4541-93df-7b71f74748fb)

New **CM v4.0 + L5 Null**:
![L1_CM40_L5-NULL](https://github.com/user-attachments/assets/ab789bb1-8f35-47a9-adc1-14894bab3c87)

#### 2. `plot --plot-type l2`

New **CM v2.9 + L5 Standard** (flat chart, no manual trims):
![L2_CM29_L5-STANDARD](https://github.com/user-attachments/assets/db3d6690-4785-42c5-8a23-d263be8cc9aa)

New **CM v4.0 + L5 Variable**:
![L2_CM40_L5-VARIABLE](https://github.com/user-attachments/assets/a27bcf45-f473-4a90-b948-281f33cdf100)

New **CM v4.0 + L5 Standard**:
![L2_CM40_L5-STANDARD](https://github.com/user-attachments/assets/10e90706-56c6-4fd8-a4be-4e87fe4f0fab)

#### 3. `plot --plot-type l8t`
![L8-trims_plot](https://github.com/user-attachments/assets/19d715f3-ad54-4342-9166-719865bc720f)

#### 4. `plot --plot-type l8s`
![L8-saturation_plot](https://github.com/user-attachments/assets/ebfeb453-85cd-419c-a333-1358d5a04200)

#### 5. `plot --plot-type l8h`
![L8-hue_plot](https://github.com/user-attachments/assets/01b88caf-36a4-402f-8ad0-0cbe06dcb9e3)

manuelrn commented:
@bbeny123 thanks again for also adding plot for L8 metadata.

One last request:
Would it be possible to add in the top information of the plots information related to L9 metadata?
I mean something similar to what is shown in the plots generated with the DoVi_Scripts tool of R3S3t9999:
![image](https://github.com/user-attachments/assets/1ba98acd-916d-4f83-8cf1-3601ad110f49)

I also noticed that in the examples you show for Summary (`info -s`), if it is CM v4.0 it shows the L2/L8 trims but if it is CM v2.9 it does not show any trims.
I guess ideally when it is CM v2.9 it should also show the L2 trims.

quietvoid commented on src/dovi/rpu_info.rs:
```diff
@@ -43,6 +55,24 @@ pub struct SummaryL1Stats {
+    pub ms_weight: (f64, f64, f64),
+}
+
+pub struct SummaryL8Stats {
```
Might be clearer if `Vector` was in the name?

quietvoid commented on src/dovi/rpu_info.rs:
```diff
@@ -333,17 +380,242 @@ impl RpusListSummary {
+                power: Self::min_max_avg(l8_data, |e| e.trim_power as f64),
+                chroma: Self::min_max_avg(l8_data, |e| e.trim_chroma_weight as f64),
+                saturation: Self::min_max_avg(l8_data, |e| e.trim_saturation_gain as f64),
+                ms_weight: Self::min_max_avg(l8_data, |e| e.ms_weight as f64),
```
Probably should include `target_mid_contrast` and `clip_trim` for L8.
Would be `Option`s  in the same trims struct.

quietvoid commented:
> Regarding the L9 metadata in the top information of the plots — I can definitely add that, as long as @quietvoid is okay with it.

It seems fine to me however L9 is also dynamic.

bbeny123 commented:
I’ve added the L9 metadata to the top information of the plots. If it’s dynamic, it will be displayed in the format:
`<MDP-alias> (<number-of-frames>), <MDP2-alias> (<number-of-frames>)`

Example (static L9):
![L8-trims_plot](https://github.com/user-attachments/assets/facfede9-f701-4662-9d5c-18d87d21df1a)

quietvoid commented on src/dovi/rpu_info.rs:
```diff
@@ -164,12 +209,16 @@ impl RpusListSummary {
+            (
+                Some((dmv1_count, dmv2_count)),
+                "1 + 2 (CM 2.9 and 4.0)",
+                false,
```
Maybe do `dmv2_count > dmv1_count` ?

quietvoid commented on src/dovi/rpu_info.rs:
```diff
@@ -333,17 +382,248 @@ impl RpusListSummary {
+            ("TOP", |l5| l5.active_area_top_offset),
+            ("BOTTOM", |l5| l5.active_area_bottom_offset),
+            ("LEFT", |l5| l5.active_area_left_offset),
+            ("RIGHT", |l5| l5.active_area_right_offset),
```
I would rather see a format like `top=280, bottom=280, left=0, right=0` 
It's more readable.

Examples of the cases:
```
top=280, bottom=280, left=0, right=0 # default case
top=N/A, bottom=N/A, left=N/A, right=N/A # no L5
top=0..280, bottom=0..280, left=0, right=0 # min max range
```

Not sure about the range format but I like it.

quietvoid commented on src/dovi/rpu_info.rs:
```diff
@@ -333,17 +382,248 @@ impl RpusListSummary {
+            .iter()
+            .map(|rpu| {
+                if let Some(ExtMetadataBlock::Level2(l2)) =
+                    rpu.vdr_dm_data.as_ref().and_then(|dm| dm.get_block(2))
```
I'm pretty sure `get_block` only returns the first trim block.
There needs to be a way to specify the target nits but ATM I'm not sure the crate has a convenient API for it.

It's probably fine for now.

bbeny123 commented:
Changes:
- Fixed info formatting for L5 offsets
- Added support for excluding series in L2/L8 trims plots
- L2/L8 trims plots can now be generated for 100/300/600/1000 nits target (default: 100 nits)
